sycl/source/detail/platform_impl.cpp

```diff
@@ -49,6 +52,113 @@ platform_impl_host::get_devices(info::device_type dev_type) const {
   return res;
 }
 
+struct DevDescT {
+  const char *devName = nullptr;
+  int devNameSize = 0;
+
+  const char *devDriverVer = nullptr;
+  int devDriverVerSize = 0;
+};
+
+static std::vector<DevDescT> getWhiteListDesc() {
+  const char *str = SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get();
+  if (!str)
+    return {};
+
+  std::vector<DevDescT> decDescs;
+  const char devNameStr[] = "DeviceName";
+  const char driverVerStr[] = "DriverVersion";
+  decDescs.emplace_back();
+  while ('\0' != *str) {
+    const char **valuePtr = nullptr;
+    int *size = nullptr;
+
+    // -1 to avoid comparing null terminator
+    if (0 == strncmp(devNameStr, str, sizeof(devNameStr) - 1)) {
+      valuePtr = &decDescs.back().devName;
+      size = &decDescs.back().devNameSize;
+      str += sizeof(devNameStr) - 1;
+    } else if (0 == strncmp(driverVerStr, str, sizeof(driverVerStr) - 1)) {
+      valuePtr = &decDescs.back().devDriverVer;
+      size = &decDescs.back().devDriverVerSize;
+      str += sizeof(driverVerStr) - 1;
+    }
+
+    if (':' != *str)
+      throw sycl::runtime_error("Malformed device white list");
+
+    // Skip ':'
+    str += 1;
+
+    if ('{' != *str || '{' != *(str + 1))
+      throw sycl::runtime_error("Malformed device white list");
+
+    // Skip opening sequence "{{"
+    str += 2;
+
+    *valuePtr = str;
+
+    // Increment until closing sequence is encountered
+    while (('\0' != *str) && ('}' != *str || '}' != *(str + 1)))
+      ++str;
+
+    if ('\0' == *str)
+      throw sycl::runtime_error("Malformed device white list");
+
+    *size = str - *valuePtr;
+
+    // Skip closing sequence "}}"
+    str += 2;
+
+    if ('\0' == *str)
+      break;
+
+    // '|' means that the is another filter
+    if ('|' == *str)
+      decDescs.emplace_back();
+    else if (',' != *str)
+      throw sycl::runtime_error("Malformed device white list");
+
+    ++str;
+  }
```

Comment:
Is it possible to rewrite this loop in something higher-level so it is more understandable?

Reply:
Will try.